riscv/csrs.cc

```diff
@@ -405,9 +407,15 @@ base_status_csr_t::base_status_csr_t(processor_t* const proc, const reg_t addr):
   has_page(proc->extension_enabled_const('S') && proc->supports_impl(IMPL_MMU)),
   sstatus_write_mask(compute_sstatus_write_mask()),
   sstatus_read_mask(sstatus_write_mask | SSTATUS_UBE | SSTATUS_UXL
+                    | (proc->extension_enabled('X') ? SSTATUS_XS : 0)
```

Comment:
This code is executed only at reset and therefore assumes that `proc->extension_enabled('X')` cannot change during runtime. Therefore this should use `proc->extension_enabled_const('X')` to document and verify that assumption.

Reply:
That is, assuming that dynamically setting `misa.X=0` should result in XS always reading 0? I'm not sure if that is allowed/required by the privileged spec or not.

If XS is allowed to be nonzero even after setting `misa.X=0` then this code is correct.

Reply:
This modification is the same as the original design, which uses any_cunstom_extension() in write_mask.
And in the misa_mask the X bit is not allowed to modify during runtime.

Reply:
The misa `write_mask` can change, and the purpose of `extension_enabled_const()` is to make sure such changes do not subtly break assumptions made elsewhere in the code.

The old `any_custom_extensions()` (which seems is dead now, and can be removed?) did not consider `misa.X` so didn't need to think about this.

The only line I can find in the spec about this is:

> In systems without additional user extensions requiring new state, the XS field is read-only zero.

And regarding bits in `misa` generally:

> When a standard extension is disabled by clearing its bit in misa, the instructions and CSRs defined or modified by the extension revert to their defined or reserved behaviors as if the extension is not implemented.

Which suggests to me that if `misa.X` is dynamically updated to 0, the XS field should also read as 0.

There's several options for achieving that, if we ever wanted to make `misa.X` writable, but at the very least we need to document the assumption here that `misa.X` is not writable, by using `extension_enabled_const()`.